### PR TITLE
Use dedicated keyring for docker packages

### LINKED ISSUE #3481: apt-key deprecation warning during installation
Hi,

it seem that the installation of docker is still using the old `apt-key` mechanism which leads to some of the following messages

```shell
W: https://download.docker.com/linux/ubuntu/dists/noble/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.

W: https://download.docker.com/linux/ubuntu/dists/jammy/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
```

The important part should be:

```shell
++ lsb_release -si
+ sudo apt-key add -
++ tr '[:upper:]' '[:lower:]'
+ curl -fsSL https://download.docker.com/linux/ubuntu/gpg
```

I found it in code at: [render.go#L58](https://github.com/kubermatic/kubeone/blob/4b759c03133e1dda5d01ee6a2b867377b1919f71/pkg/scripts/render.go#L58)

As it is shown from the official docs it should be done via, see [here](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository):

```shell
sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
sudo chmod a+r /etc/apt/keyrings/docker.asc

# Add the repository to Apt sources:
echo \
  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
  $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
```

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
apt-key shows a deprecation warning

**Which issue(s) this PR fixes**:

Fixes #3481

**What type of PR is this?**
/kind cleanup
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Use dedicated keyring for docker packages
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Comment by xmudrii** [comment]
/release-note-edit
```release-note
Use dedicated keyring for Docker repositories to solve `apt-key` deprecation warning upon installing/upgrading containerd
```

File: pkg/scripts/render.go [hunk 1]
```diff
@@ -54,9 +54,12 @@ var containerRuntimeTemplates = map[string]string{
 			{{ if .CONFIGURE_REPOSITORIES }}
 			sudo apt-get update
 			sudo apt-get install -y apt-transport-https ca-certificates curl software-properties-common lsb-release
-			curl -fsSL https://download.docker.com/linux/$(lsb_release -si | tr '[:upper:]' '[:lower:]')/gpg |
-				sudo apt-key add -
-			sudo add-apt-repository "deb https://download.docker.com/linux/$(lsb_release -si | tr '[:upper:]' '[:lower:]') $(lsb_release -cs) stable"
```
Comment (xmudrii) [inline pkg/scripts/render.go:59]: Why did we replace `$(lsb_release -si | tr '[:upper:]' '[:lower:]')` with `ubuntu`? That was a little neat trick that allowed us to use KubeOne with both Ubuntu and Debian, even though Debian isn't officially supported.

Comment (kron4eg) [reply pkg/scripts/render.go:59]: It took it from our Ubuntu OSP.

Comment (xmudrii) [reply pkg/scripts/render.go:59]: Created https://github.com/kubermatic/kubeone/pull/3508 and https://github.com/kubermatic/operating-system-manager/pull/431 to address this comment

**Pull request merged**